tests/test_enterprise/api/test_serializers.py

```diff
@@ -395,3 +397,37 @@ def test_update_enterprise_courses(self, mock_config_helpers, _):
         # Make sure global context passed in to update_course is added to the course.
         assert all('tpa_hint' in course for course in self.serializer.data['results'])
         assert all(course['tpa_hint'] == self.provider_id for course in self.serializer.data['results'])
+
+
+@ddt.ddt
+@mark.django_db
+class TestEnterpriseCustomerUserWriteSerializer(APITest):
+    """
+    Tests for the ``EnterpriseCustomerUserWriteSerializer``.
+    """
+
+    @ddt.data(
+        (TEST_USERNAME, 201),
+        ('non-existing-username', 400),
+    )
+    @ddt.unpack
+    @override_settings(ECOMMERCE_SERVICE_WORKER_USERNAME=TEST_USERNAME)
+    def test_validate_username(self, username, expected_status_code):
+        """
+        Success for POSTing with users (determined by username) depends on whether the user exists.
+        """
+        self.user.is_staff = True
+        self.user.save()
+        permission = Permission.objects.get(name='Can add Enterprise Customer Learner')
+        self.user.user_permissions.add(permission)
```

Comment:
Yeah this is handled by the `DjangoModelPermissions` that we added, rather than by us. It will check for permissions for certain unsafe methods like `POST`, `DELETE`, etc.. This is something @douglashall recommended and I really like the idea.